src/js/models/documentindex.js

```diff
@@ -0,0 +1,200 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Immutable = require("immutable");
+
+    var object = require("js/util/object");
+
+    /**
+     * Model of the Photoshop document index.
+     * 
+     * @constructor
+     */
+    var DocumentIndex = Immutable.Record({
+        /**
+         * An ordered list of document IDs
+         * @type {Immutable.List.<number>}
+         */
+        openDocumentIDs: Immutable.List(),
+
+        /**
+         * The index of the currently active document; negative if there are none
+         * @type {?number}
+         */
+        activeDocumentIndex: -1,
+
+        /**
+         * The ID of the currently active document, or null if there are none
+         * @type {?number}
+         */
+        activeDocumentID: null
+    });
+
+    /**
+     * Set the position of the given document ID in the document index.
+     *
+     * @private
+     * @param {number} documentID
+     * @param {number} position
+     * @return {DocumentIndex}
+     */
+    DocumentIndex.prototype._updateDocumentPosition = function (documentID, position) {
+        // find the document in the array of indices
+        var openDocumentIDs = this.openDocumentIDs,
+            currentIndex = openDocumentIDs.indexOf(documentID),
+            nextDocumentIDs = openDocumentIDs;
+        
+        if (currentIndex > -1) {
+            // remove it from the index
+            nextDocumentIDs = nextDocumentIDs.delete(currentIndex);
+        }
+
+        // re-add it at the correct position
+        nextDocumentIDs = nextDocumentIDs.splice(position, 0, documentID);
+
+        return this.set("openDocumentIDs", nextDocumentIDs);
+    };
+
+    /**
+     * Set the active document in the index.
+     *
+     * @param {number} documentID
+     * @param {number=} position
+     * @return {DocumentIndex}
+     */
+    DocumentIndex.prototype.setActive = function (documentID, position) {
+        if (position === undefined) {
+            position = this.openDocumentIDs.indexOf(documentID);
+            if (position < 0) {
+                throw new Error("Document ID not found in index: " + documentID);
+            }
+        }
+
+        return this.merge({
+            activeDocumentID: documentID,
+            activeDocumentIndex: position
+        });
+    };
+
+    /**
+     * Set (or reset) the position of a document in the index.
+     * 
+     * @param {number} documentID
+     * @param {number} position
+     * @param {boolean=} active
+     * @return {DocumentIndex}
+     */
+    DocumentIndex.prototype.setPosition = function (documentID, position, active) {
+        var nextIndex = this._updateDocumentPosition(documentID, position);
+
+        if (active) {
+            nextIndex = nextIndex.setActive(documentID, position);
```

Comment:
I think that's the intended behavior. It only updates the active ID and index during setPosition if `active === true`.